mopidy/core/library.py

```diff
@@ -236,7 +236,7 @@ def lookup(self, uri=None, uris=None):
                 result = future.get()
                 if result is not None:
                     validation.check_instances(result, models.Track)
-                    results[u] = result
+                    results[u] = [r for r in result if r.uri]
```

Comment:
Maybe add a note or TODO to revisit this if we make the URI mandatory. So maybe just `# Silently drop tracks without URIs as this crashes some clients`. And on a side note should we log what backend sent the bad data?